lib/bundler/cli/outdated.rb

```diff
@@ -30,6 +30,13 @@ def run
         Bundler.definition(:gems => gems, :sources => sources)
       end
 
+      patch_level = [:major, :minor, :patch].select {|v| options.keys.include?(v.to_s) }
+      Bundler.definition.create_gem_version_promoter([]).tap do |gvp|
+        gvp.level = patch_level.first || :major
+        gvp.strict = options[:strict]
+      end
+      Bundler.load
```

Comment:
also, @ryanfox1985, after all this typing about _my_ stuff, it may make sense to revert this PR to _just_ the grouping code you're trying to contribute. I may be overloading this PR with one too many things.